tests/unit/configuration_test.py

```diff
@@ -3095,3 +3098,141 @@ def test_default_workers_per_api_key(self):
+                {"path": "path1"},
+                {"path": "path2", "api_key_id": "second_key"},
+                {"path": "path3", "api_key_id": "key3"},
+            ],
+        }
+
+        log_config2 = {"path": "path5", "api_key_id": "key3"}
+
+        config_fragment = {
+            "api_keys": [
+                {"api_key": "another_key3", "id": "key3"},
+                {"api_key": "another_key4", "workers": 3},
+            ],
+            "logs": [
+                {"path": "path4"},
+                log_config2,
+                {"path": "path6", "api_key_id": "second_key"},
+            ],
+        }
+        self._write_file_with_separator_conversion(json.dumps(main_config))
+        self._write_config_fragment_file_with_separator_conversion(
+            "a.json", json.dumps(config_fragment)
+        )
+
+        config = self._create_test_configuration_instance()
+        config.parse()
+
+        api_keys = list(config.api_key_configs)
+
+        assert len(api_keys) == 5
+
+        assert api_keys[0]["id"] == "default"
+        assert api_keys[1]["id"] == "second_key"
+        assert api_keys[2]["id"] == "2"
+        assert api_keys[3]["id"] == "key3"
+        assert api_keys[4]["id"] == "4"
+
+        log_config2["api_key_id"] = "wrong_api_key_id"
+        self._write_file_with_separator_conversion(json.dumps(main_config))
+        self._write_config_fragment_file_with_separator_conversion(
+            "a.json", json.dumps(config_fragment)
+        )
+
+        config = self._create_test_configuration_instance()
+
+        with pytest.raises(BadConfiguration) as err_info:
+            config.parse()
+
+        assert (
+            "The log file '/var/log/scalyr-agent-2/path5' entry refers to a non-existing api key with id 'wrong_api_key_id'"
+            in err_info.value.message
+        )
+
+    def test_api_keys_negative_workers_number(self):
+        self._write_file_with_separator_conversion(
+            """ {
+            api_key: "hi there"
+
+            api_keys: [
+                {"api_key": "another_key", "id": "second_key"},
+                {"api_key": "another_key2", "workers": -1}
+            ]
+          }
+        """
+        )
+
+        config = self._create_test_configuration_instance()
+        with pytest.raises(BadConfiguration) as err_info:
+            config.parse()
+
+        assert "Value must be greater than 1" in err_info.value.message
+
+        self._write_file_with_separator_conversion(
+            """ {
+            api_key: "hi there"
+            default_workers_per_api_key: -1,
+            api_keys: [
+                {"api_key": "another_key", "id": "second_key"},
+                {"api_key": "another_key2"}
+            ]
+          }
+        """
+        )
+
+        config = self._create_test_configuration_instance()
+        with pytest.raises(BadConfiguration) as err_info:
+            config.parse()
+
+        assert "Value must be greater than 1" in err_info.value.message
```

Comment:
Oops, I think configuration.py needs to be updated to say "Value must be greater than or equal to ...". Same goes for "max_value" (it's just a bug / typo in that error message).

Reply:
64db8892